comictalker/talkers/comicvine.py

```diff
@@ -439,7 +440,7 @@ def format_issue_results(
 
             formatted_results.append(
                 ComicIssue(
-                    aliases=record["aliases"],
+                    aliases=record["aliases"].split("\n") if record["aliases"] else [],
```

Comment:
Should this be done in comiccacher and anywhere a possible empty list is to be split to prevent the [''] issue?

Reply:
This is a separate issue that CV has, aliases instead of either not being there or being an empty list it is null.
To fix the issue of ending up with a list with a single empty element you can just use `.strip()` before the split

Reply:
Sadly strip won't help. When I had the problem before I used the answer here: https://stackoverflow.com/questions/10780423/split-function-avoid-last-empty-space
Before the alt images was changed:
```
url_list = [x for x in issue["alt_images_url"].split(",") if x]
```
For now in `format_search_results` I've done:
```
            if record["aliases"]:
                aliases = record["aliases"].split("\n")
            else:
                aliases = []
```
Maybe there is a better way and/or add a function to utils to do this?

Reply:
> Sadly strip won't help. When I had the problem before I used the answer here: https://stackoverflow.com/questions/10780423/split-function-avoid-last-empty-space Before the alt images was changed:
> 
> ```
> url_list = [x for x in issue["alt_images_url"].split(",") if x]
> ```

Yes it does, using the example in the linked question, what it doesn't fix is a duplicated separator in the middle, which I am not interested in fixing if CV leaves a new line in the middle then that is a valid alias:
```pycon
>>> s = 'James;Joseph;Arun;'
>>> s.split(';')
['James', 'Joseph', 'Arun', '']
>>> 
>>> s.strip(';').split(';')
['James', 'Joseph', 'Arun']
>>> 
>>> 
>>> s = 'James;Joseph;;Arun;'
>>> s.strip(';').split(';')
['James', 'Joseph', '', 'Arun']
>>> 
>>> [x for x in s.split("\n") if x]
['James;Joseph;;Arun;']
>>> 
>>> 
>>> s = '''
... James
... Joseph
... Arun
... '''
>>> s.split("\n")
['', 'James', 'Joseph', 'Arun', '']
>>> 
>>> s.strip("\n").split("\n")
['James', 'Joseph', 'Arun']
>>> 
>>> 
>>> s = '''
... James
... Joseph
... 
... Arun
... '''
>>> s.strip("\n").split("\n")
['James', 'Joseph', '', 'Arun']
>>> 
>>> [x for x in s.split("\n") if x]
['James', 'Joseph', 'Arun']
```

> For now in `format_search_results` I've done:
> 
> ```
>             if record["aliases"]:
>                 aliases = record["aliases"].split("\n")
>             else:
>                 aliases = []
> ```
> 
> Maybe there is a better way and/or add a function to utils to do this?

This is fine but this is unrelated to the issue with the extra newline (or comma)

Reply:
I meant generally as any empty string will produce [''] and what to use for comiccacher or do we accept that an empty field will return ['']? Should the tests expect [] or ['']?

Reply:
So I was about to link to the documentation for [split] but instead I'm linking to the documentation for [splitlines]
```pycon
>>> s = '''
... James
... Joseph
... Arun
... '''
>>> s.splitlines()
['', 'James', 'Joseph', 'Arun']
>>> 
>>> s.strip("\n").splitlines()
['James', 'Joseph', 'Arun']
>>> 
>>> 
>>> s = '''
... James
... Joseph
... 
... Arun
... '''
>>> s.strip("\n").splitlines()
['James', 'Joseph', '', 'Arun']
>>> 
>>> [x for x in s.splitlines() if x]
['James', 'Joseph', 'Arun']
>>> 
>>> s = ""
>>> 
>>> s.splitlines()
[]
```

Just change it to `.strip().splitlines()` and call it done

[split]: https://docs.python.org/3/library/stdtypes.html#str.split
[splitlines]: https://docs.python.org/3/library/stdtypes.html#str.splitlines